### PR TITLE
Validate  BA109

### PR DESCRIPTION
**Pull request by YairGlik**
## Related Issues
fixes:https://jira-dc.paloaltonetworks.com/browse/CIAC-9836

### CODE REVIEW

File: demisto_sdk/commands/validate/initializer.py [hunk 1]
```diff
@@ -526,7 +526,10 @@ def get_items_status(
                 elif self.is_code_file(path, path_str):
                     path = self.obtain_yml_from_code(path_str)
                     if path not in statuses_dict:
-                        statuses_dict[path] = git_status
+                        if git_status != GitStatuses.RENAMED:
+                            statuses_dict[path] = git_status
+                        else:
+                            statuses_dict[path] = None
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/initializer.py:532]: why this change?

File: demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            if valid_base_name != base_name:
+                invalid_files.append(file_name)
+                valid_files.append(valid_base_name.join(file_name.rsplit(base_name, 1)))
+
+        if invalid_files:
+            self.invalid_files = invalid_files
+            self.valid_files = valid_files
+            return True
+
+        return False
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py:91]: Let's save all the results in one list containing tuples of (valid_file, invalid file)
Also, return it from **check_separators_in_files** instead of a boolean flag and remove the new fields added to the class.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Nice work!
See my comments.

File: demisto_sdk/commands/validate/initializer.py (hunk 1 above)
Comment (YairGlik) [reply demisto_sdk/commands/validate/initializer.py:532]: This is an addition you wrote because of an error I received

File: demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py (hunk 2 above)
Comment (YairGlik) [reply demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py:91]: I hope that's what you meant

**Review by YuvHayun — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+class FileNameHasSeparatorsValidator(BaseValidator[ContentTypes]):
+    error_code = "BA109"
+    description = (
+        "Check if there are separators in the script or integration files names."
+    )
+    rationale = "Filenames for scripts and integrations should not contain separators to maintain consistency and readability."
+    error_message = "The {item_type} files should be named without any separators in the base name:\n{params}"
+    related_field = "file path"
+    is_auto_fixable = False
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py:26]: Let's add gitstatuses: added

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 4]
```diff
@@ -1945,6 +1948,47 @@ def test_IsContentItemNameContainTrailingSpacesValidator_fix(
...
+            create_script_object(name="TestScript"),
+            0,
+            [""],
+            id="valid script",
+        ),
+    ],
+)
+def test_FileNameHasSeparatorsValidator_is_valid(
+    content_item, expected_number_of_failures, expected_messages
+):
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:1982]: Add given-when-then doc strgins.

**Review by YuvHayun — APPROVED** [review]

File: demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py (hunk 3 above)
Comment (YairGlik) [reply demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py:26]: i added:
    expected_git_statuses = [
        GitStatuses.RENAMED,
        GitStatuses.ADDED,
    ]
